src/ui/control/scale_control.js

```diff
@@ -52,40 +53,47 @@ class ScaleControl {
         return 'bottom-left';
     }
 
-    _onMove() {
-        updateScale(this._map, this._container, this.options);
+    _onUpdate() {
+        updateScale(this._map, this._container, this._language, this.options);
     }
 
     onAdd(map: Map): HTMLElement {
         this._map = map;
+        this._language = map.getLanguage();
         this._container = DOM.create('div', 'mapboxgl-ctrl mapboxgl-ctrl-scale', map.getContainer());
+        this._container.dir = 'auto';
```

Comment:
Should this be set with a `direction` CSS property instead?

Reply:
Sure 👍

Reply:
However, it seems that CSS `direction` property doesn't support the `auto` detection, and we should set the direction explicitly